### PR TITLE
Opentsdb alerting feature completed, ready for testing

### LINKED ISSUE #6270: Alerting: Support for opentsdb in alerting
We need help form the community if we want to support opentsdb in grafana 4.0

@utkarshcmu have started working on this and it would be great of more opentsdb users would help out to test it. Support for opentsdb should be added to the `alerting_opentsdb` branch.

ref https://github.com/grafana/grafana/pull/6269
ref https://github.com/grafana/grafana/issues/5858

### PR DESCRIPTION
**Pull request by utkarshcmu**
This resolves #6270 and partially #5858 and makes OpenTSDB alerting ready for testing by the community.

### CODE REVIEW

File: pkg/tsdb/opentsdb/opentsdb.go [hunk 1]
```diff
@@ -58,14 +57,64 @@ func (e *OpenTsdbExecutor) Execute(ctx context.Context, queries tsdb.QuerySlice,
...
+				downsampleInterval = "1m"  //default value for blank
+			}
+			downsample :=  downsampleInterval + "-" + queries[i].Model.Get("downsampleAggregator").MustString()
+			if queries[i].Model.Get("downsampleFillPolicy").MustString() != "none" {
+				metric["downsample"] = downsample + "-" + queries[i].Model.Get("downsampleFillPolicy").MustString()
+			}
 		}
+
+		// Setting rate options
+		if queries[i].Model.Get("shouldComputeRate").MustBool() {
```
Comment (torkelo) [inline pkg/tsdb/opentsdb/opentsdb.go]: some go tests for this would be nice, see the Prometheus implementation for reference :)

**Comment by utkarshcmu** [comment]
Added tests :)

File: pkg/tsdb/opentsdb/opentsdb.go [hunk 2]
```diff
@@ -58,14 +57,11 @@ func (e *OpenTsdbExecutor) Execute(ctx context.Context, queries tsdb.QuerySlice,
...
-	for _, query := range queries {
-		tsdbQuery.Queries = []OpenTsdbMetric{
-			OpenTsdbMetric{
-				Metric:     query.Model.Get("metric").MustString(),
-				Aggregator: query.Model.Get("aggregator").MustString(),
-			},
-		}
+  for i := 0; i < len(queries); i++ {
+  	metric := e.buildMetric(queries[i])
+  	tsdbQuery.Queries[i] = metric
```
Comment (bergquist) [inline pkg/tsdb/opentsdb/opentsdb.go]: Any reason for not appending metric to tsdbQuery.Queries? Ex `tsdbQuery.Queries = append(tsdbQuery.Queries, metric) `

**Review by bergquist — COMMENTED** [review]
Two minor comments. Looks good otherwise.

**Comment by utkarshcmu** [comment]
Fixed the for loop and append function as suggested by @bergquist. Ready for merge now. Excited to see alerting for OpenTSDB in Grafana. Waiting for the build once this PR will be merged to master. :)

**Pull request merged**